public_root_of_trust.go

```diff
@@ -0,0 +1,93 @@
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"os"
+	"time"
+
+	sigtuf "github.com/sigstore/sigstore/pkg/tuf"
+	tufclient "github.com/theupdateframework/go-tuf/client"
+)
+
+var globalBuffer bytes.Buffer
+
+type inMemoryDest struct{}
+
+func (d inMemoryDest) Write(p []byte) (n int, err error) {
+	return globalBuffer.Write(p)
+}
+
+func (d inMemoryDest) Delete() error {
+	panic("inMemoryDest delete function should not run")
+}
+
+func GetTargets(usage sigtuf.UsageKind, proxy Proxy) ([]sigtuf.TargetFile, error) {
+	// client initialization
+	var httpClient *http.Client
+	if proxy.URL != "" {
+		transport := proxy.HttpTransport()
+		httpClient = &http.Client{
+			Transport: &transport,
+			Timeout:   20 * time.Second,
+		}
+	}
+	remoteStore, err := tufclient.HTTPRemoteStore(sigtuf.DefaultRemoteRoot, nil, httpClient)
```

Comment:
when proxy.URL is "", httpClient is nil and 
tufclient.HTTPRemoteStore(sigtuf.DefaultRemoteRoot, nil, nil) is called.
In tufclient.HTTPRemoteStore() it then uses http.DefaultClient as the http client object.
But http.DefaultClient is actually Default{}.
There are already many posts on the internet about "[Don't use Go's default HTTP client (in production)]"
Please do not pass nil as httpClient .

Reply:
Ok. The primary concern being that there is no timeout length set with the default client. I have updated this so that a timeout length is set whether or not a user has a proxy configured.